pumps/segment.go

```diff
@@ -0,0 +1,99 @@
+package pumps
+
+import (
+	"fmt"
+	"encoding/json"
+	"errors"
+	"github.com/Sirupsen/logrus"
+	segment "github.com/Segment/analytics-go"
+)
+
+type SegmentPump struct {
+	segmentClient *segment.Client
+	segmentConf   *SegmentConf
+}
+
+var segmentPrefix string = "segment-pump"
+
+type SegmentConf struct {
+	WriteKey string `mapstructure:"segment_write_key"`
+}
+
+func (s *SegmentPump) New() Pump {
+	newPump := SegmentPump{}
+	return &newPump
+}
+
+func (s *SegmentPump) GetName() string {
+	return "Segment Pump"
+}
+
+func (s *SegmentPump) Init(config interface{}) error {
+	s.segmentConf = &SegmentConf{}
+	loadConfigErr := mapstructure.Decode(config, &s.segmentConf)
+
+	if loadConfigErr != nil {
+		log.WithFields(logrus.Fields{
+			"prefix": segmentPrefix,
+		}).Fatal("Failed to decode configuration: ", loadConfigErr)
+	}
+
+	s.segmentClient = segment.New(s.segmentConf.WriteKey)
+
+	return nil
+}
+
+func (s *SegmentPump) WriteData(data []interface{}) error {
+	log.WithFields(logrus.Fields{
+		"prefix": segmentPrefix,
+	}).Info("Writing ", len(data), " records")
+
+	for i, v := range data {
+		analyticsRecord := AnalyticsRecord{}
+		err := msgpack.Unmarshal(v.([]byte), &analyticsRecord)
```

Comment:
Does this actually work?? You shouldn't need to unmarshal the data using msgpack, it's already unmarshalled when it hits this handler. There is a special case for uptime data, which issn't handled in other pumps. See the [influx handler](https://github.com/TykTechnologies/tyk-pump/blob/develop/pumps/influx.go#L92) for proof ;-)